plugins/single_plugins/fast-switcher.cpp

```diff
@@ -8,23 +8,56 @@
 #include <wayfire/view.hpp>
 #include <wayfire/workspace-manager.hpp>
 #include <wayfire/util/log.hpp>
+#include <wayfire/render-manager.hpp>
+#include <wayfire/util/duration.hpp>
 
 /*
  * This plugin provides abilities to switch between views.
  * It works similarly to the alt-esc binding in Windows or GNOME
  */
 
+class animated_alpha_transformer : public wf::scene::view_2d_transformer_t
```

Comment:
Do you actually have to subclass this for some reason or can you just use a wf::scene::view_2d_transformer_t as it is, [like in animate](https://github.com/WayfireWM/wayfire/blob/master/plugins/animate/basic_animations.hpp#L33)?

Reply:
Possibly not, I might have done this by getting inspiration from another plugin. Thanks for the tip, I will look.

Reply:
So in the end I need a way to store an instance of `wf::animation::simple_animation_t` per-view. In the case of basic_animations there is already a subclass of the per-view `animation_base` that stores it, but there is no such equivalent in the fast-switcher plugin, so I don't see a simple way of getting rid of this subclass.

Reply:
Like soreau, I don't think inheritance makes sense here. Better simply store a custom_data_t on the view with the animation running (and don't inherit from view_2d_transformer)